### ISSUE TITLE
Type Error when trying to follow the example

### ISSUE THREAD

**Issue by typkrft** [python-caldav/caldav#141]
When I try to use 

```PYTHON
## Let's try to find or create a calendar ...
try:
    ## This will raise a NotFoundError if calendar does not exist
    my_new_calendar = my_principal.calendar(name="Test calendar")
    assert(my_new_calendar)
    ## calendar did exist, probably it was made on an earlier run
    ## of this script
except caldav.error.NotFoundError:
    ## Let's create a calendar
    my_new_calendar = my_principal.make_calendar(name="Test calendar")
```
in my code it throws the error `TypeError: quote_from_bytes() expected bytes`.

Surely I'm doing something wrong, but I'm not sure what. Here's my code trying to roughly mirror the example. 

```PYTHON
from time import sleep
from datetime import datetime
import uuid
import requests
import json
import caldav

def init_tasklist(tasklist_name, caldav_url, username, password):
    client = caldav.DAVClient(url=caldav_url, username=username, password=password)
    my_principal = client.principal()
    calendars = my_principal.calendars()

    try:
        tasklist = my_principal.calendar(name=tasklist_name)
        assert(tasklist)
    except caldav.error.NotFoundError:
        tasklist = my_principal.make_calendar(
        name=tasklist_name, supported_calendar_component_set=['VTODO'])

init_tasklist('Test', 'https://sub.domain.com', 'myuser', 'secretpassword')
```

**Comment by tobixen** [comment]
No, I think this is a bug and not an error at your side.  Are you using python2?  At some point I stopped running tests for python2, after all it is officially end of life.

The error message in itself does not give me enough information to pinpoint it, didn't you get a traceback?

Try putting a b before quotes to force it into a byte string ... like this: `my_principal.calendar(name=b'Test calendar')`

**Comment by typkrft** [comment]
Hey thanks for the response. 

``` PYTHON
Import sys
print(sys.version) 

# Output
# 3.9.2 (default, Feb 24 2021, 13:26:09) 
# [Clang 12.0.0 (clang-1200.0.32.29)]
```
I tried using `my_principal.calendar(name=b'Test calendar')` where `Test calendar` explicitly refers to my calendar, but I got the same type error. I also tried to convert the variable to bytes prior to referencing it, to no avail.

**Comment by tobixen** [comment]
I also managed to run the tests with python2 now.

I need ...

* The full traceback (I'm quite sure there should be some more information, identifying where the exception was thrown)
* The version of caldav you're using (0.8.0 was released previous week.  0.8.1 will be released as soon as this bug is weeded out).

**Comment by typkrft** [comment]
```PYTHON
Traceback (most recent call last):
  File "/var/folders/hz/6k_nrprd29l_qj9z72t1ybf00000gn/T/tempCodeRunnerFile.python", line 20, in <module>
    init_tasklist('REDACTED', 'https://REDACTED', 'REDACTED', 'REDACTED')
  File "/var/folders/hz/6k_nrprd29l_qj9z72t1ybf00000gn/T/tempCodeRunnerFile.python", line 14, in init_tasklist
    tasklist = my_principal.calendar(name=tasklist_name)
  File "/usr/local/lib/python3.9/site-packages/caldav/objects.py", line 400, in calendar
    return self.calendar_home_set.calendar(name, cal_id)
  File "/usr/local/lib/python3.9/site-packages/caldav/objects.py", line 348, in calendar
    url=self.url.join(quote(cal_id)), id=cal_id)
  File "/usr/local/Cellar/python@3.9/3.9.2_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/urllib/parse.py", line 862, in quote
    return quote_from_bytes(string, safe)
  File "/usr/local/Cellar/python@3.9/3.9.2_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/urllib/parse.py", line 887, in quote_from_bytes
    raise TypeError("quote_from_bytes() expected bytes")
TypeError: quote_from_bytes() expected bytes
```

``` PYTHON
Name: caldav
Version: 0.7.1
Summary: CalDAV (RFC4791) client library
Home-page: https://github.com/python-caldav/caldav
Author: Cyril Robert
Author-email: cyril@hippie.io
License: GPL
Location: /usr/local/lib/python3.9/site-packages
Requires: lxml, requests, six, vobject
Required-by:
```
I didn't realize i wasn't on the latest I just installed it with pip roughly a week ago. I must have installed just right before you released 0.8.0. So sorry for the inconvenience. This seems to work as expected with 0.8.0. I will close this, but keep you informed should any other issue arise.

**Comment by tobixen** [comment]
Still strange that this didn't work with 0.7.1, but happy that upgrading to 0.8 solved the issue :-)
